### PR TITLE
Feature/azure container

### LINKED ISSUE #3: determine a better way to share secrets with the container
Currently, the `.env` file with secrets (which a user needs to create locally) is [copied into the Docker image](https://github.com/RMI-PACTA/workflow.data.preparation/blob/021ee8bed5970f797768a46378621c367f1c2651/Dockerfile#L5) (along with everything else). The GitHub PAT is necessary so the pkg install commands in the Dockerfile can [install our private packages](https://github.com/RMI-PACTA/workflow.data.preparation/blob/021ee8bed5970f797768a46378621c367f1c2651/Dockerfile#L11-L14). The [database credentials are needed during runtime](https://github.com/RMI-PACTA/workflow.data.preparation/blob/021ee8bed5970f797768a46378621c367f1c2651/run_pacta_data_preparation.R#L37-L38) in an instance so that it can connect to our FactSet data.

Although we have no intention of sharing a Docker image created with this repo with anyone, it would be ideal to find a way to not burn these secrets into the image.

### LINKED ISSUE #17: Write STDOUT log to file

### PR DESCRIPTION
**Pull request by AlexAxthelm**
Contains infrastrucure to Build a docker image suitable to run on Azure Container Instances, as well as a deploy template.

Coincidentally:
Closes #3 
Closes #17

### CODE REVIEW

File: config.yml [hunk 1]
```diff
@@ -94,3 +95,11 @@ default:
   scenario_sources_list: ["GECO2022", "IPR2021", "ISF2021", "WEO2022"]
   scenario_raw_data_to_include: ["geco_2022", "ipr_2021", "isf_2021", "weo_2022"]
   global_aggregate_scenario_sources_list: ["WEO2022"]
+
+2022Q4_CICD:
```
Comment (cjyetman) [inline config.yml:99]: what does CICD stand for?

File: run_pacta_data_preparation.R [hunk 2]
```diff
@@ -165,7 +175,7 @@ log_info("Scraping index regions... ")
 index_regions <- pacta.data.scraping::get_index_regions()
 
 
-# pull factset data ------------------------------------------------------------
+ pull factset data ------------------------------------------------------------
```
Comment (cjyetman) [inline run_pacta_data_preparation.R:178]: seems like this was inadvertently uncommented?

File: config.yml (hunk 1 above)
Comment (AlexAxthelm) [reply config.yml:99]: Continuous Integration/Continuous Deployment

Comment (AlexAxthelm) [reply config.yml:99]: I'm not married to the name

Comment (cjyetman) [reply config.yml:99]: Just curious

**Comment by jdhoffa** [comment]
I think I'm just out of sync from being away, or maybe you and CJ already discussed this in person, but could you ELI5 this in the PR description?

I'm having a hard time following what this does. 

That said, will try to test it in my machine later today, I wanted to try to get data prep running anyway to make sure I can still do it 😀

File: run_pacta_data_preparation.R [hunk 3]
```diff
@@ -1,31 +1,39 @@
...
   library(dplyr)
   library(readr)
   library(rlang)
   library(RSQLite)
   library(stringr)
   library(tidyr)
-
-  # used for logging
-  library(rlog)
-  if (interactive()) Sys.setenv("LOG_LEVEL" = "ERROR")
```
Comment (cjyetman) [inline run_pacta_data_preparation.R:18]: replace with `library(logger)`?

**Comment by cjyetman** [comment]
I feel like completely swapping out one dependency for another ({rlog} to {logger}) should be in a separate PR, not one titled "Feature/azure container".

**Comment by AlexAxthelm** [comment]
@jdhoffa still a fair bit of testing to do on my side, but while it's cooking, I'll be writing docs for it.

**Comment by jdhoffa** [comment]
Also agree with CJ. Very happy to swap rlog with logger, but prob deserves it's own PR.

File: run_pacta_data_preparation.R (hunk 3 above)
Comment (AlexAxthelm) [reply run_pacta_data_preparation.R:18]: I've made all the logging calls explicit with `logger::log_info()` (and friends).

**Comment by AlexAxthelm** [comment]
Switching to `logger` available as #76

**Comment by cjyetman** [comment]
@AlexAxthelm is this superseded by one or more of the other recent PRs? can/should we close it?

**Comment by AlexAxthelm** [comment]
Closing this, but not deleting the branch yet, since there's some elements from that that I'd like to bring into `main`, I just need to get some time to do that.
